src/algorand/AlgorandEngine.ts

```diff
@@ -95,6 +106,152 @@ export class AlgorandEngine extends CurrencyEngine<
     }
     this.minimumAddressBalance = this.networkInfo.minimumAddressBalance
     this.totalReserve = this.minimumAddressBalance
+
+    this.otherMethods = {
+      // Wallet Connect utils
+      wcInit: async (
+        wcProps: WcProps,
+        walletName: string = 'Edge'
+      ): Promise<WcDappDetails> => {
+        return await timeout(
+          // eslint-disable-next-line @typescript-eslint/no-unused-vars
+          new Promise((resolve, reject) => {
+            const connector = new WalletConnect({
+              uri: wcProps.uri,
+              storageId: wcProps.uri,
+              clientMeta: {
+                description: 'Edge Wallet',
+                url: 'https://www.edge.app',
+                icons: ['https://content.edge.app/Edge_logo_Icon.png'],
+                name: walletName
+              }
+            })
+
+            connector.on(
+              'session_request',
+              (error: Error | null, payload: AlgoWcRpcPayload) => {
+                if (error != null) {
+                  this.error(`Wallet connect session_request`, error)
+                  throw error
+                }
+
+                const params = asWcSessionRequestParams(payload).params[0]
+                const dApp = { ...params, timeConnected: Date.now() / 1000 }
+
+                // Set connector in memory
+                this.tools.walletConnectors[wcProps.uri] = {
+                  connector,
+                  wcProps,
+                  dApp
+                }
+                resolve(dApp)
+              }
+            )
+
+            // Subscribe to call requests
+            connector.on(
+              'call_request',
+              (error: Error | null, payload: AlgoWcRpcPayload) => {
+                try {
+                  if (error != null) throw error
```

Comment:
This line seems weird to me. If we throw the error, it will go back into the wallet-connect SDK. Then what? Does the SDK swallow it, or does it come out somewhere else? Do we want to show it to the user?

Same for the other 4 callbacks in this block.

Reply:
I don't have a good answer for this as it was copied from the ETH implementation and I haven't seen an error come through before. We'd have to create a special return type for the gui to parse and I don't think it's worthwhile if the v1 functionality is set to sunset in 2 months. The gui-first implementation of v2 should include this kind of error handling

Reply:
Fair enough.